### ISSUE TITLE
Py4JJavaError: An error occurred while calling z:com.johnsnowlabs.nlp.pretrained.PythonResourceDownloader.downloadPipeline.

### ISSUE THREAD

**Issue by herokyar** [JohnSnowLabs/spark-nlp#478]
Hi, 

I am running pyspark 2.2 (local pip install)  with spark-nlp 1.8.3 (local pip install)

Below is how I create the SparkSession.

```

## create spark object with sparknlp library
spark = SparkSession \
    .builder \
    .appName("Python Spark SQL SDD") \
    .config("spark.jars.packages","JohnSnowLabs:spark-nlp:1.8.3") \
    .getOrCreate()

```

But below code is failing:

```
# then, test the library with BasicPipeline
from sparknlp.pretrained.pipeline.en import BasicPipeline
#Annotate with pipeline
BasicPipeline().annotate("Please parse this sentence. Thanks")
BasicPipeline().annotate(["This is a first sentence", "This is another one"])
#Just get the pipeline
BasicPipeline().pretrained()
```

Below is the error:

---------------------------------------------------------------------------
Py4JJavaError                             Traceback (most recent call last)
<ipython-input-10-7568dfc1f56e> in <module>()
      2 from sparknlp.pretrained.pipeline.en import BasicPipeline
      3 #Annotate with pipeline
----> 4 BasicPipeline().annotate("Please parse this sentence. Thanks")
      5 BasicPipeline().annotate(["This is a first sentence", "This is another one"])
      6 #Just get the pipeline

/Users/infy/anaconda/envs/tc-env/lib/python2.7/site-packages/sparknlp/pretrained/pipeline/en.pyc in annotate(target, target_column)
     23     @staticmethod
     24     def annotate(target, target_column=None):
---> 25         return _Model.annotate(BasicPipeline, "pipeline_basic", target, target_column)
     26 
     27     @staticmethod

/Users/infy/anaconda/envs/tc-env/lib/python2.7/site-packages/sparknlp/pretrained/pipeline/en.pyc in annotate(model_class, name, target, target_column)
      8     def annotate(model_class, name, target, target_column=None):
      9         if not model_class.model:
---> 10             model_class.model = ResourceDownloader().downloadPipeline(name, "en")
     11         if type(target) is pyspark.sql.dataframe.DataFrame:
     12             if not target_column:

/Users/infy/anaconda/envs/tc-env/lib/python2.7/site-packages/sparknlp/pretrained/__init__.pyc in downloadPipeline(name, language, remote_loc)
     36     @staticmethod
     37     def downloadPipeline(name, language, remote_loc=None):
---> 38         j_obj = _internal._DownloadPipeline(name, language, remote_loc).apply()
     39         jmodel = JavaModel()
     40         jmodel._java_obj = j_obj

/Users/infy/anaconda/envs/tc-env/lib/python2.7/site-packages/sparknlp/internal.pyc in __init__(self, name, language, remote_loc)
     58     def __init__(self, name, language, remote_loc):
     59         super(_DownloadPipeline, self).__init__("com.johnsnowlabs.nlp.pretrained.PythonResourceDownloader.downloadPipeline")
---> 60         self._java_obj = self._new_java_obj(self._java_obj, name, language, remote_loc)
     61 
     62 

/Users/infy/anaconda/envs/tc-env/lib/python2.7/site-packages/pyspark/ml/wrapper.pyc in _new_java_obj(java_class, *args)
     61             java_obj = getattr(java_obj, name)
     62         java_args = [_py2java(sc, arg) for arg in args]
---> 63         return java_obj(*java_args)
     64 
     65     @staticmethod

/Users/infy/anaconda/envs/tc-env/lib/python2.7/site-packages/py4j/java_gateway.pyc in __call__(self, *args)
   1131         answer = self.gateway_client.send_command(command)
   1132         return_value = get_return_value(
-> 1133             answer, self.gateway_client, self.target_id, self.name)
   1134 
   1135         for temp_arg in temp_args:

/Users/infy/anaconda/envs/tc-env/lib/python2.7/site-packages/pyspark/sql/utils.pyc in deco(*a, **kw)
     61     def deco(*a, **kw):
     62         try:
---> 63             return f(*a, **kw)
     64         except py4j.protocol.Py4JJavaError as e:
     65             s = e.java_exception.toString()

/Users/infy/anaconda/envs/tc-env/lib/python2.7/site-packages/py4j/protocol.pyc in get_return_value(answer, gateway_client, target_id, name)
    317                 raise Py4JJavaError(
    318                     "An error occurred while calling {0}{1}{2}.\n".
--> 319                     format(target_id, ".", name), value)
    320             else:
    321                 raise Py4JError(

Py4JJavaError: An error occurred while calling z:com.johnsnowlabs.nlp.pretrained.PythonResourceDownloader.downloadPipeline.
: java.lang.NoSuchMethodError: org.json4s.jackson.JsonMethods$.parse$default$3()Z
	at com.johnsnowlabs.nlp.pretrained.ResourceMetadata$.parseJson(ResourceMetadata.scala:53)
	at com.johnsnowlabs.nlp.pretrained.ResourceMetadata$$anonfun$readResources$1.apply(ResourceMetadata.scala:82)
	at com.johnsnowlabs.nlp.pretrained.ResourceMetadata$$anonfun$readResources$1.apply(ResourceMetadata.scala:81)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at scala.collection.generic.Growable$class.$plus$plus$eq(Growable.scala:59)
	at scala.collection.mutable.ListBuffer.$plus$plus$eq(ListBuffer.scala:183)
	at scala.collection.mutable.ListBuffer.$plus$plus$eq(ListBuffer.scala:45)
	at scala.collection.TraversableOnce$class.to(TraversableOnce.scala:310)
	at scala.collection.AbstractIterator.to(Iterator.scala:1336)
	at scala.collection.TraversableOnce$class.toList(TraversableOnce.scala:294)
	at scala.collection.AbstractIterator.toList(Iterator.scala:1336)
	at com.johnsnowlabs.nlp.pretrained.ResourceMetadata$.readResources(ResourceMetadata.scala:84)
	at com.johnsnowlabs.nlp.pretrained.ResourceMetadata$.readResources(ResourceMetadata.scala:76)
	at com.johnsnowlabs.nlp.pretrained.S3ResourceDownloader.downloadMetadataIfNeed(S3ResourceDownloader.scala:67)
	at com.johnsnowlabs.nlp.pretrained.S3ResourceDownloader.resolveLink(S3ResourceDownloader.scala:77)
	at com.johnsnowlabs.nlp.pretrained.S3ResourceDownloader.download(S3ResourceDownloader.scala:89)
	at com.johnsnowlabs.nlp.pretrained.ResourceDownloader$.downloadResource(ResourceDownloader.scala:98)
	at com.johnsnowlabs.nlp.pretrained.ResourceDownloader$.downloadPipeline(ResourceDownloader.scala:130)
	at com.johnsnowlabs.nlp.pretrained.ResourceDownloader$.downloadPipeline(ResourceDownloader.scala:125)
	at com.johnsnowlabs.nlp.pretrained.PythonResourceDownloader$.downloadPipeline(ResourceDownloader.scala:190)
	at com.johnsnowlabs.nlp.pretrained.PythonResourceDownloader.downloadPipeline(ResourceDownloader.scala)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:280)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:214)
	at java.base/java.lang.Thread.run(Thread.java:834)

I was able to run below query with success though .

`advancedPipeline = PipelineModel.load("/Users/infy/Downloads/pipeline_basic_en_1/")`

Thanks.

**Comment by dekelcohen** [comment]
Similar error with spark 2.3.x and sprak-nlp 2.0.1

Which spark versions does spark-nlp 2.0.1 supports ?
Which tensorflow versions ?

**Comment by maziyarpanahi** [comment]
@dekelcohen For your issue and Apache Spark support you should read this: https://github.com/JohnSnowLabs/spark-nlp#apache-spark-support

However, you do not need to even have TensorFlow if you are not training NER. So the TF version is irrelevant. 
For more examples or testing Spark NLP 2.0.1 without installing anything (Docker) you can check this repo: https://github.com/JohnSnowLabs/spark-nlp-workshop

@herokyar I think basic and advanced pipelines are removed from auto download in favor of other pipelines: https://github.com/JohnSnowLabs/spark-nlp#models-and-pipelines
However, you can still download them and manually load them.

**Comment by nehadave78** [comment]
Hi @maziyarpanahi  how can I manually load it in Python? I was trying https://github.com/JohnSnowLabs/spark-nlp-workshop/blob/master/jupyter/quick_start.ipynb 
and getting above error. I downloaded 'entity_recognizer_dl' in my local. But how do I load it?

**Comment by maziyarpanahi** [comment]
Hi @nehadave78, if you are using pretrained() you don't need to load it, it all happens automatically. But if you need to download it manually and then load it because your Spark setup doesn't have internet then you can take a look at the documentation:
https://nlp.johnsnowlabs.com/docs/en/pipelines
Also, please beware of Spark and Spark NLP versions. (latest is `2.0.8` and supports Spark `2.4.3`)

**Comment by nehadave78** [comment]
Thanks JohnSnowLabs/spark-nlp.
I am using pyspark to run using notebook.so it should have internet. But it
gives error

**Comment by maziyarpanahi** [comment]
@nehadave78 could you please create new issue with some codes that regenerates the error + Spark and Spark NLP?
Thank you

**Comment by huglittlecat88** [comment]
Hello, I had a problem to be solved. It troubled me lots.

python 3.7 
tensorflow 1.13 
spark-nlp: 2.2.1
spark: 2.4.0 

Using PySpark , win10 

Py4JJavaError: An error occurred while calling z:com.johnsnowlabs.nlp.pretrained.PythonResourceDownloader.downloadPipeline.
: java.lang.UnsupportedOperationException: Spark NLP tried to load a Tensorflow Graph using Contrib module, but failed to load it on this system. If you are on Windows, this operation is not supported. Please try a noncontrib model. If not the case, please report this issue. Original error message:

Op type not registered 'BlockLSTM' in binary running on LAPTOP-O2LQN8MT. Make sure the Op and Kernel are registered in the binary running in this process. Note that if you are loading a saved graph which used ops from tf.contrib, accessing (e.g.) `tf.contrib.resampler` should be done before importing the graph, as contrib ops are lazily registered when the module is first accessed.
	at com.johnsnowlabs.ml.tensorflow.TensorflowWrapper$.readGraph(TensorflowWrapper.scala:163)
	at com.johnsnowlabs.ml.tensorflow.TensorflowWrapper$.read(TensorflowWrapper.scala:202)
	at com.johnsnowlabs.ml.tensorflow.ReadTensorflowModel$class.readTensorflowModel(TensorflowSerializeModel.scala:73)
	at com.johnsnowlabs.nlp.annotators.ner.dl.NerDLModel$.readTensorflowModel(NerDLModel.scala:139)
	at com.johnsnowlabs.nlp.annotators.ner.dl.ReadsNERGraph$class.readNerGraph(NerDLModel.scala:117)
	at com.johnsnowlabs.nlp.annotators.ner.dl.NerDLModel$.readNerGraph(NerDLModel.scala:139)
	at com.johnsnowlabs.nlp.annotators.ner.dl.ReadsNERGraph$$anonfun$2.apply(NerDLModel.scala:121)
	at com.johnsnowlabs.nlp.annotators.ner.dl.ReadsNERGraph$$anonfun$2.apply(NerDLModel.scala:121)
	at com.johnsnowlabs.nlp.ParamsAndFeaturesReadable$$anonfun$com$johnsnowlabs$nlp$ParamsAndFeaturesReadable$$onRead$1.apply(ParamsAndFeaturesReadable.scala:31)
	at com.johnsnowlabs.nlp.ParamsAndFeaturesReadable$$anonfun$com$johnsnowlabs$nlp$ParamsAndFeaturesReadable$$onRead$1.apply(ParamsAndFeaturesReadable.scala:30)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at com.johnsnowlabs.nlp.ParamsAndFeaturesReadable$class.com$johnsnowlabs$nlp$ParamsAndFeaturesReadable$$onRead(ParamsAndFeaturesReadable.scala:30)
	at com.johnsnowlabs.nlp.ParamsAndFeaturesReadable$$anonfun$read$1.apply(ParamsAndFeaturesReadable.scala:41)
	at com.johnsnowlabs.nlp.ParamsAndFeaturesReadable$$anonfun$read$1.apply(ParamsAndFeaturesReadable.scala:41)
	at com.johnsnowlabs.nlp.FeaturesReader.load(ParamsAndFeaturesReadable.scala:19)
	at com.johnsnowlabs.nlp.FeaturesReader.load(ParamsAndFeaturesReadable.scala:8)
	at org.apache.spark.ml.util.DefaultParamsReader$.loadParamsInstance(ReadWrite.scala:652)
	at org.apache.spark.ml.Pipeline$SharedReadWrite$$anonfun$4.apply(Pipeline.scala:274)
	at org.apache.spark.ml.Pipeline$SharedReadWrite$$anonfun$4.apply(Pipeline.scala:272)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:186)
	at org.apache.spark.ml.Pipeline$SharedReadWrite$.load(Pipeline.scala:272)
	at org.apache.spark.ml.PipelineModel$PipelineModelReader.load(Pipeline.scala:348)
	at org.apache.spark.ml.PipelineModel$PipelineModelReader.load(Pipeline.scala:342)
	at com.johnsnowlabs.nlp.pretrained.ResourceDownloader$.downloadPipeline(ResourceDownloader.scala:360)
	at com.johnsnowlabs.nlp.pretrained.ResourceDownloader$.downloadPipeline(ResourceDownloader.scala:354)
	at com.johnsnowlabs.nlp.pretrained.PythonResourceDownloader$.downloadPipeline(ResourceDownloader.scala:446)
	at com.johnsnowlabs.nlp.pretrained.PythonResourceDownloader.downloadPipeline(ResourceDownloader.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:748)

**Comment by sdeepmars** [comment]
> Hi @nehadave78, if you are using pretrained() you don't need to load it, it all happens automatically. But if you need to download it manually and then load it because your Spark setup doesn't have internet then you can take a look at the documentation:
> https://nlp.johnsnowlabs.com/docs/en/pipelines
> Also, please beware of Spark and Spark NLP versions. (latest is `2.0.8` and supports Spark `2.4.3`)

but the "offline" section in 'https://nlp.johnsnowlabs.com/docs/en/pipelines' do not have any sample of how to. I tired giving the path tot he downloaded model folder(which has two folders inside) it gives error like load cant have two params. Can you suggest any document for loading locally downloaded model. Thank you

**Comment by maziyarpanahi** [comment]
Hi @sdeepmars 
That section used to have an example, not sure what happened to it. Here there is an example at the bottom: https://github.com/JohnSnowLabs/spark-nlp#pipelines

It's simple, download the pipeline, extract it, use the path that is acceptable by Apache Spark (local or cluster).
